internal/provider/resource_tfe_workspace.go

```diff
@@ -41,16 +41,6 @@ func resourceTFEWorkspace() *schema.Resource {
 		},
 
 		CustomizeDiff: func(c context.Context, d *schema.ResourceDiff, meta interface{}) error {
-			// NOTE: execution mode and agent_pool_id must be set to default first before calling
-			// the validation functions
-			if err := setComputedDefaults(c, d); err != nil {
-				return err
-			}
-
-			if err := overwriteDefaultExecutionMode(c, d); err != nil {
-				return err
-			}
-
```

Comment:
It does seem to be a tradeoff when extracting this argument into a separate resource.  Removing the computed default does break the unsetting case out of necessity because otherwise it would continuously conflict when you unset it to add `tfe_workspace_settings`.

I highlighted this as a breaking change in the release notes, but it deserves more attention, so I added a note in the documentation.